### ISSUE TITLE
No such file or directory

### ISSUE THREAD

**Issue by d4f5409d** [ArielOSProject/AuroraStore#3]
When I download apps, this happens:

Error: no such file or directory
I have all the permissions enabled for Aurora Store
Device: Samsung Galaxy S8
Store version: latest

Error log:
java.io.FileNotFoundException: open failed: ENOENT (No such file or directory)
	at android.os.ParcelFileDescriptor.openInternal(ParcelFileDescriptor.java:313)
	at android.os.ParcelFileDescriptor.open(ParcelFileDescriptor.java:211)
	at androidx.core.content.FileProvider.openFile(:2)
	at android.content.ContentProvider.openAssetFile(ContentProvider.java:1566)
	at android.content.ContentProvider.openTypedAssetFile(ContentProvider.java:1746)
	at android.content.ContentProvider.openTypedAssetFile(ContentProvider.java:1812)
	at android.content.ContentProvider$Transport.openTypedAssetFile(ContentProvider.java:425)
	at android.content.ContentResolver.openTypedAssetFileDescriptor(ContentResolver.java:1465)
	at android.content.ContentResolver.openAssetFileDescriptor(ContentResolver.java:1302)
	at android.content.ContentResolver.openInputStream(ContentResolver.java:1022)
	at e.b.a.q.g.h.a(:9)
	at e.b.a.a.a.g.h.a(:1)
	at e0.a.a.l0$a.a(:1)
	at e0.a.a.y$a.run(:2)
	at e0.a.a.p0.l.run(Unknown Source:9)
	at java.lang.Thread.run(Thread.java:764)

**Comment by zrhoffman** [comment]
This happens to me, most recently today (currently using 4.0.2 on AOSiP 10-Gapps-hotdogb-2021-03-16). When it happens, I remove the app's cache and storage and set up Aurora store. Then the error goes away for awhile.
![Screenshot of *No such file or directory*](https://user-images.githubusercontent.com/11163823/113445010-40780000-93b2-11eb-8dbd-34c294a17d18.png)

<details><summary>Edit: Click to show stacktrace, since mine has slightly different line numbers than @d4f5409d's</summary>

```java
java.io.FileNotFoundException: open failed: ENOENT (No such file or directory)
 at android.os.ParcelFileDescriptor.openInternal(ParcelFileDescriptor.java:319)
 at android.os.ParcelFileDescriptor.open(ParcelFileDescriptor.java:224)
 at androidx.core.content.FileProvider.openFile(:2)
 at android.content.ContentProvider.openAssetFile(ContentProvider.java:1712)
 at android.content.ContentProvider.openTypedAssetFile(ContentProvider.java:1894)
 at android.content.ContentProvider.openTypedAssetFile(ContentProvider.java:1961)
 at android.content.ContentProvider$Transport.openTypedAssetFile(ContentProvider.java:511)
 at android.content.ContentResolver.openTypedAssetFileDescriptor(ContentResolver.java:1689)
 at android.content.ContentResolver.openAssetFileDescriptor(ContentResolver.java:1505)
 at android.content.ContentResolver.openInputStream(ContentResolver.java:1189)
 at e.b.a.q.g.h.a(:9)
 at e.b.a.a.a.g.h.a(:1)
 at e0.a.a.l0$a.a(:1)
 at e0.a.a.y$a.run(:2)
 at e0.a.a.p0.l.run(Unknown Source:9)
 at java.lang.Thread.run(Thread.java:919)
 ```
 </details>

**Comment by d4f5409d** [comment]
@zrhoffman Firefox Nightly gang😁

**Comment by zrhoffman** [comment]
Aurora Store is on GitLab now. *File Not Found Error on Update - open failed: ENOENT* issue: https://gitlab.com/AuroraOSS/AuroraStore/-/issues/562

**Comment by mikalackis** [comment]
@zrhoffman  @d4f5409d  this repo is a fork of Aurora Store and was not updated for quite some time now. The gitlab link in the previous comment is the up-to-date and official repo so thanks @zrhoffman for posting it.

Related to the issue you guys are having, I experienced similar on my side and clearing the downloads list always solves the issue. For what its worth, i'm also using Aurora Services as an installer.

**Comment by zrhoffman** [comment]
Thanks for explaining. FWIW, [gitlab.com/AuroraOSS/AuroraStore](https://gitlab.com/AuroraOSS/AuroraStore) fixed this issue in 4.0.3, so no need for ArielOSProject/AuroraStore#3 to stay open IMO.
